kafka-connect-elastic6/src/main/scala/com/datamountaineer/streamreactor/connect/elastic6/ElasticJsonWriter.scala

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2017 Datamountaineer.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.datamountaineer.streamreactor.connect.elastic6
+
+import java.util
+
+import com.datamountaineer.kcql.{Kcql, WriteModeEnum}
+import com.datamountaineer.streamreactor.connect.converters.FieldConverter
+import com.datamountaineer.streamreactor.connect.elastic6.config.ElasticSettings
+import com.datamountaineer.streamreactor.connect.elastic6.indexname.CreateIndex
+import com.datamountaineer.streamreactor.connect.errors.ErrorHandler
+import com.datamountaineer.streamreactor.connect.schemas.ConverterUtil
+import com.fasterxml.jackson.databind.JsonNode
+import com.landoop.sql.Field
+import com.sksamuel.elastic4s.ElasticDsl._
+import com.sksamuel.elastic4s.Indexable
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import org.apache.kafka.common.config.ConfigException
+import org.apache.kafka.connect.sink.SinkRecord
+import org.elasticsearch.action.support.WriteRequest.RefreshPolicy
+
+import scala.collection.JavaConversions._
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
+import scala.util.Try
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class ElasticJsonWriter(client: KElasticClient, settings: ElasticSettings)
+  extends ErrorHandler with StrictLogging with ConverterUtil {
+
+  logger.info("Initialising Elastic Json writer")
+
+  //initialize error tracker
+  initialize(settings.taskRetries, settings.errorPolicy)
+
+  //create the index automatically if it was set to do so
+  settings.kcqls.filter(_.isAutoCreate).foreach(client.index)
+
+  settings.kcqls.filter(_.getWriteMode == WriteModeEnum.UPSERT).foreach { kcql =>
+    if (kcql.getPrimaryKeys.size() != 1) {
```

Comment:
should we allow multiple PK fields and they form the ES key by combining them witha  a separator? (separator can be `.`  - configurable)?

Reply:
Maybe to the same for es5? @Antwnis @andrewstevenson

Reply:
Hi @stheppi, I will work for this.

Reply:
Can I change es5 too?

Reply:
@caiooliveiraeti Sure do es5 :-)